packages/mobx/src/types/observablemap.ts

```diff
@@ -351,7 +351,7 @@ export class ObservableMap<K = any, V = any>
             } else if (Array.isArray(other)) {
                 other.forEach(([key, value]) => this.set(key, value))
             } else if (isES6Map(other)) {
-                if (other.constructor !== Map) {
+                if (other.constructor.name !== "Map") {
```

Comment:
Hm, I can still name my subclass `Map`, what about:
`Object.getPrototypeOf(Object.getPrototypeOf(Object.getPrototypeOf(other))) === null`
or
`Object.prototype.toString.call(Object.getPrototypeOf(Object.getPrototypeOf(other))) === '[object Object]'`

Reply:
Oh, I see. I added `isPlainES6Map()` to check if a map instance is strictly plain Map object. 103d4b03